pkg/integrations/blackbox_exporter/blackbox_exporter.go

```diff
@@ -0,0 +1,162 @@
+package blackbox_exporter
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/agent/pkg/integrations"
+	"github.com/grafana/agent/pkg/integrations/config"
+	blackbox_config "github.com/prometheus/blackbox_exporter/config"
+)
+
+// DefaultConfig holds the default settings for the blackbox_exporter integration.
+var DefaultConfig = Config{
+	BlackboxConfigFile: "",
+	BlackboxTargets:    make([]BlackboxTarget, 0),
+}
+
+func LoadFile(filename string, log log.Logger) (*blackbox_config.Config, error) {
+	sc := &blackbox_config.SafeConfig{
+		C: &blackbox_config.Config{},
+	}
+	err := sc.ReloadConfig(filename, log)
+	if err != nil {
+		return nil, err
+	}
+	return sc.C, nil
+}
+
+//go:generate curl https://raw.githubusercontent.com/prometheus/blackbox_exporter/v0.20.0/example.yml --output blackbox.yml
+
+func LoadEmbeddedConfig(log log.Logger) (*blackbox_config.Config, error) {
+	dir, err := os.Getwd()
```

Comment:
This feels like it will fail when running as a windows service. Unless Go is doing some magic on detecting running as a service.

Reply:
It actually doesn't work in linux either. Any ideas how can I make this work? I wanna curl the file and get the path of the file to embed in blackbox config

Reply:
Instead of go:generate, I would embed the file and leave instructions on where you got it from.